### PR TITLE
Upgrade to ruby 3

### PR DESCRIPTION
**Pull request by wendy-clio**
Upgrade to ruby 3.1 
---
From [another PR](https://github.com/clio/jit_preloader/pull/69) with a CI outcome that enables deprecation warnings, we can see that only tests with Gemfile 5.2(ActiveRecord ~>5.2) has warnings on the method within the ActiveModel gem, which comes with ActiveRecord.

After checking the[ ruby/rail support list](https://www.fastruby.io/blog/ruby/rails/versions/compatibility-table.html), Rail version < 6.1 is no longer supported and not compatible with ruby version >= 3.0.0.  So, I take this PR a chance to also remove unsupported Rail versions (5.2 & 6.0).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,7 +22,7 @@ jobs:
       - name: Set up Ruby ${{ matrix.ruby-version }}
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: 2.7
+          ruby-version: 3.0
```
Comment (peerkleio) [inline .github/workflows/ci.yml:25]: If you can go to 3.1 that would be even better!

**Review by peerkleio — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -14,8 +14,6 @@ jobs:
       matrix:
         gemfile:
           - Gemfile
-          - Gemfile.5.2
```
Comment (wendy-clio) [inline .github/workflows/ci.yml:17]: Removing CI job for unsupported rail versions.

File: Gemfile [hunk 3]
```diff
@@ -1,5 +1,6 @@
 source 'https://rubygems.org'
 
 gem "activerecord", ">=7"
+gem "sqlite3", "~> 1.4"
```
Comment (wendy-clio) [inline Gemfile:4]: This is a constraint on ActiveRecord connection https://github.com/rails/rails/blob/v7.1.3.4/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L14
Without it, we will get higher version of sqlite3 which break the connection.
[Latest beta version](https://github.com/rails/rails/blob/v7.2.0.beta2/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L14) already fixed the problem, we just need to wait until the stable version to upgrade.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (wendy-clio) [reply .github/workflows/ci.yml]: done!

**Review by tudorbertiean — APPROVED** [review]

**Review by anthonyoconnorclio — APPROVED** [review]

**Review by Vtown121 — APPROVED** [review]

**Review by gavingmiller — APPROVED** [review]

**Pull request merged**